### ISSUE TITLE
Avoid returning errors where possible 

### ISSUE THREAD

**Issue by e00E** [rust-fuzz/arbitrary#82]
https://github.com/rust-fuzz/arbitrary/commit/1d2f65360f634968b5525f4d6430a88a376277f0

> Next breaking release that we make, I think we can remove the fallibility from
the `Arbitrary` trait and all the `Unstructured` methods.

This wasn't addressed in the discussion about 1.0 https://github.com/rust-fuzz/arbitrary/issues/62 . 

Is this still something we want to do? Are you open for a PR implementing it?

**Comment by fitzgen** [comment]
I don't think it is worth addressing at this time, because we just cut 1.0 and I don't think it is worth cutting a 2.0 just for this change.

Really unfortunate that this slipped through the cracks though :(
